### PR TITLE
Docs/use nbqa on notebooks

### LINKED ISSUE #1004: Use nbqa on notebooks
@carlocagnetta Could you look into this? Should be hopefully a low hanging fruit

https://nbqa.readthedocs.io/en/latest/readme.html#readme

### PR DESCRIPTION
**Pull request by carlocagnetta**
- [ ] I have added the correct label(s) to this Pull Request or linked the relevant issue(s)
- [x] I have provided a description of the changes in this Pull Request
- [x] I have added documentation for my changes
- [x] If applicable, I have added tests to cover my changes.
- [x] I have reformatted the code using `poe format` 
- [x] I have checked style and types with `poe lint` and `poe type-check`
- [x] (Optional) I ran tests locally with `poe test` 
(or a subset of them with `poe test-reduced`) ,and they pass
- [x] (Optional) I have tested that documentation builds correctly with `poe doc-build`

# PR Description

- Added nbqa to poetry env
- Resolved mypy issues on notebooks and related files
- Conducting ruff checks on notebooks
- Add DataclassPPrintMixin for better stats representation
- Improved Notebooks wording and explanations

Resolve: #1004, #974

### CODE REVIEW

**Comment by carlocagnetta** [comment]
@MischaPanch I would like to add nbqa ruff check and nbqa mypy to the pre-commit-config

**Comment by MischaPanch** [comment]
> @MischaPanch I would like to add nbqa ruff check and nbqa mypy to the pre-commit-config

Sure, go ahead :)

File: pyproject.toml [hunk 1]
```diff
@@ -178,13 +179,15 @@ test = "pytest test --cov=tianshou --cov-report=xml --cov-report=term-missing --
 test-reduced = "pytest test/base test/continuous --cov=tianshou --durations=0 -v --color=yes"
 _black_check = "black --check ."
 _ruff_check = "ruff check ."
+_ruff_check_nb = "nbqa ruff docs/02_notebooks"
```
Comment (MischaPanch) [inline pyproject.toml]: This is problematic b/c it's used in `poe lint` which is not supposed to format anything. However, this command actually formats the notebooks (even without the `--fix`). I know that `nbqa "ruff --check ."` didn't work for a mysterious reason, but I wouldn't want to modify the logic of lint. 

If you don't find a fix for  `nbqa "ruff --check ."`, we could introduce a new poe command `nb-lint-format` and call in in CI additionally to `poe lint`. It's not nice, but better than changing `lint` semantics. Then pls open an issue in the nbqa repository describing the problem and add a block comment here that links to the issue

**Review by MischaPanch — CHANGES_REQUESTED** [review]
Only the minor change in pyproject.toml is needed. You can also add the nbqa stuff to pre-commit. Then this is ready for merging, thanks @carlocagnetta !

**Review by MischaPanch — APPROVED** [review]

**Pull request merged**